config/rbac/kustomization.yaml

```diff
@@ -5,35 +5,44 @@ resources:
   - submariner-operator/role_binding.yaml
   - submariner-operator/cluster_role.yaml
   - submariner-operator/cluster_role_binding.yaml
+  - submariner-operator/privileged_cluster_role.yaml
```

Comment:
We are not removing the existing cluster_role/role_binding yamls. Can you please elaborate the plan?

Reply:
Hmm, good point.
A. We need to keep the existing cluster role to allow submariner component watch/create/etc k8s resources , check [0] for LH agent for example.

B. In addition, to allow privileged permission we need to create a clusterrole , like [1] and for each submariner component have clusterrolebinding resource pointing to this resource like [2] .

Initially, I thought that we can only  create a new clusterrole (like [1]) and add another roleref to existing clusterrolebinding (like 3) , but IIUC we can only define single roleref in clusterrolebinding (check 4).

so, IIUC, we have two options here:
1. Don't create any new files, add new rule (allowing access to privileged scc) in existing clusterrolebinding  of each SM componenet.
2. Create seperate files for SCC permission.

I decided to go with option 2, because didn't want to tight privileged SCC permission to other API permissions, but I'm totally open to new ideas.


[0]
https://github.com/submariner-io/submariner-operator/blob/devel/config/rbac/lighthouse-agent/cluster_role.yaml
[1] 
https://github.com/submariner-io/submariner-operator/blob/6b15cfeedec578cbe18ec2f977000db17175bb79/config/rbac/submariner-operator/privileged_cluster_role.yaml
[2] 
https://github.com/submariner-io/submariner-operator/blob/6b15cfeedec578cbe18ec2f977000db17175bb79/config/rbac/submariner-operator/privileged_cluster_role_binding.yaml

[3] 
https://github.com/submariner-io/submariner-operator/blob/devel/config/rbac/lighthouse-agent/cluster_role_binding.yaml

[4]
https://pkg.go.dev/k8s.io/kubernetes/pkg/apis/rbac#ClusterRoleBinding

Reply:
In option '1' above, it should be 'add new rule (allowing access to privileged scc) in existing clusterrole of each SM componenet' and not clusterrolebinding.

So, for each relevant component under https://github.com/submariner-io/submariner-operator/tree/devel/config/rbac , add a new rule in clusterrole allowing privileged access.

Reply:
>In option '1' above, it should be 'add new rule (allowing access to privileged scc) in existing clusterrole of each SM componenet' and not clusterrolebinding.

Correct. So far AFAICT whenever we require any new permissions for a component, we used to update the existing ClusterRole with the "apiGroups/resources/verbs" and as you said, this does not require any changes to the ClusterRoleBinding. This looks clean and simple. However, at the time of upgrades, the operator has to check (i think it might already be doing it?) not only if the clusterRole exists, but it matches with the clusterRole definition of the new version. If there is any mis-match a new clusterRole has to be installed. 

I did not work much in this area so cannot suggest if we should go with 2 (or 1), but I was personally following 1 so far. @skitt @vthapar @tpantelis your thoughts?

Reply:
I haven't worked much in this area either, IIUC, changes are expected in the way to acquire privileged permisisons in the next versions of OCP.
So, maybe it makes sense to keep the separate API  (scc.Update API in subctl) which implies for option '2'.

Reply:
FWIW I think we should strive to be as vendor agnostic as possible, so it makes more sense to store any "extra" vendor stuff, such as the openshift specific resources, in separate files, and any generic stuff, such as the K8S RBAC, in the general files we already have.

Reply:
Yes, it makes sense not to mix K8S RBAC with SCC and apply the relevant files only when necessary

Reply:
+1 to Mike's comment. Separate files for specific vendors and single file for cmmon generic. Then update existing ClusterRoles with newer permissions. Add new file if need something specific for a vendor.

Reply:
I added separate Openshift RBAC files for each component.

A. The OCP ClusterRole and ClusterRoleBinding yaml files should
be applied by [1] PR in Subctl Repo.

B. After [1] is merged the rest of OCP specific rules should be
moved to OCP ClusterRole and ClusterRoleBinding files in a
follow up PR.



[1] https://github.com/submariner-io/subctl/pull/258